### PR TITLE
Create #704_Binary_Search.py

### LINKED ISSUE #1: #704_Binary_Search
# Summary
add 704. Bianry Search

# Goal
add 704. Bianry Search

# Todo
- [ ] add 704. Bianry Search
- [ ] time complexity: O(logn)
- [ ] space complexity O(1)

# Deadline
/

# Parent issue
# References
None

# Notes
None

### PR DESCRIPTION
**Pull request by ashizawaN**
Closes #1

### CODE REVIEW

File: #704_Binary_Search.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Solution:
+    def search(self, nums: List[int], target: int) -> int:
+        left = 0
+        right = len(nums) - 1
+        
+        while left <= right:
+            m = (left + right) // 2
```
Comment (nayuta-ai) [inline #704_Binary_Search.py:7]: Rewrite it as follows. 
```mid = (left+right) // 2```

※ Reference
https://pep8-ja.readthedocs.io/ja/latest/#section-14

File: #704_Binary_Search.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Solution:
+    def search(self, nums: List[int], target: int) -> int:
```
Comment (nayuta-ai) [inline #704_Binary_Search.py:2]: Import the List module because it is not a built-in module.

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

File: #704_Binary_Search.py (hunk 2 above)
Comment (ashizawaN) [reply #704_Binary_Search.py:2]: @nayuta-ai 
I posted the code that I submitted to LeetCode. Is this code not enough? Actually, I don't understand enough how to import a module and what is need it for.
Could you give me some hints or something helpful?

Comment (nayuta-ai) [reply #704_Binary_Search.py:2]: Trying to execute from your local environment makes you understand the issue. 

* Reference: https://docs.python.org/3.8/library/typing.html

File: #704_Binary_Search.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from typing import List
```
Comment (nayuta-ai) [inline #704_Binary_Search.py:1]: Expect two blank lines between imports and class.

File: #704_Binary_Search.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+class Solution:
+    def search(self, nums: List[int], target: int) -> int:
+        left = 0
+        right = len(nums) - 1
+        
+        while left <= right:
+            m = (left+right) // 2
```
Comment (nayuta-ai) [inline #704_Binary_Search.py:9]: "m" is an incomprehensible variable name.
Change from "m" to "mid".

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

**Review by nayuta-ai — APPROVED** [review]

**Pull request merged**